src/net_processing.cpp

```diff
@@ -153,7 +154,12 @@ namespace {
     std::atomic<int64_t> g_last_tip_update(0);
 
     /** Relay map */
-    typedef std::map<uint256, CTransactionRef> MapRelay;
+    struct RelayEntry {
+        explicit RelayEntry(CTransactionRef &&tx) : m_txref(tx) {}
+        CTransactionRef m_txref;
+        std::unordered_set<NodeId> m_node_set;
```

Comment:
There are at most 9 buckets, so a set of outbound `NodeId`s and a flag for inbound ones might demand less memory?